### PR TITLE
Date-Picker Kit

### PR DESCRIPTION
**Pull request by bh247484**
#### Screens

screenshot

#### Breaking Changes

Added an [autocomplete prop](https://github.com/powerhome/playbook/blob/9f3df7ad661dc500954d43f52d3f9a3067875df6/app/pb_kits/playbook/pb_text_input/_text_input.html.erb#L16-L17) to the text input kit.  I don't think it will be a breaking change though.  It's designed to return `nil` if the default prop value is passed so it shouldn't conflict with any existing autocomplete attributes.

#### Runway Ticket URL

[Runway Story](https://nitro.powerhrg.com/runway/backlog_items/NUX-1393)

#### Checklist:

- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review
- [x] **SCREENSHOT** Please add a screen shot or two
- [x] **SPECS** Please cover your changes with specs

### CODE REVIEW

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 1]
```diff
@@ -0,0 +1,1018 @@
...
+      top: 1px;
+      right: 0px;
+      border-left: 1px solid $border_light;
+      width: 45px;
+      height: 43.2px;
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      padding-left: 11px;
+      color: $text_lt_light;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:69]: Can we replace some of these with variables?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 2]
```diff
@@ -0,0 +1,1018 @@
...
+    .cal_icon {
+      width: 20px;
+    }
+
+    // Header Styles
+    $header_height: 40px;
+
+    .flatpickr-months {
+      // padding: 5px;
+      // padding-bottom: 5px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:83]: Remove

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 3]
```diff
@@ -0,0 +1,1018 @@
...
+      border-bottom: solid 1px $border_light;
+    }
+    .flatpickr-month {
+      height: $header_height;
+    }
+    .flatpickr-current-month {
+      left: 10px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:90]: variable?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 4]
```diff
@@ -0,0 +1,1018 @@
...
+      display: flex;
+      // justify-content: flex-start;
+      padding-top: 0px;
+    }
+    .flatpickr-monthDropdown-months {
+      appearance: none;
+      padding-left: 0px;
+      padding-right: 14px;
+      margin-left: 15px;
+      height: $header_height;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:100]: This top area is pretty strange. @kre8sions can the selects be improved with flexbox or grid?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 5]
```diff
@@ -0,0 +1,1018 @@
...
+      &:hover {
+        background-color: white;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:102]:
```suggestion
        background-color: $white;
```

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 6]
```diff
@@ -0,0 +1,1018 @@
...
+      }
+    }
+    .numInputWrapper {
+      display: contents;
+      .numInput {
+        appearance: none;
+        border-left: solid 1px $border_light;
+        padding-left: 22.5px;
+        padding-right: 14px;
+        margin-left: 10px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:112]: variables?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 7]
```diff
@@ -0,0 +1,1018 @@
...
+          cursor: pointer;
+        }
+      }
+    }
+
+    .flatpickr-prev-month {
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      left: 234px !important;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:127]: Can we use flexbox to change the position?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 8]
```diff
@@ -0,0 +1,1018 @@
...
+      padding-right: 10px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:128]: Variable

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 9]
```diff
@@ -0,0 +1,1018 @@
...
+      @include caption;
+      user-select: none;
+    }
+    // Month-Year Select Styles
+    select.flatpickr-monthDropdown-months {
+      font-size: 16px;
+      font-weight: 600;
+      color: #242B42;
+      margin-top: 0px;
+      margin-bottom: 0px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:169]: Why are these needed? We should be using our text styles. Also variables

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 10]
```diff
@@ -0,0 +1,1018 @@
...
+    }
+
+    select.numInput {
+      font-size: 16px;
+      font-weight: 600;
+      color: #242B42;
+      border: 0px;
+      font-family: inherit;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:177]: tokens

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 11]
```diff
@@ -0,0 +1,1018 @@
...
+      user-select: none;
+      &:hover {
+        background-color: rgba($color: $neutral, $alpha: 0.2);
+        border-color: rgba($color: $neutral, $alpha: 0.2);
+        box-shadow: $shadow_deep;
+      }
+    }
+    
+    .flatpickr-day.today {
+      font-weight: 600;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:196]: ??

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 12]
```diff
@@ -0,0 +1,1018 @@
...
+      color: $charcoal;
+      &.selected {
+        color: $text_dk_default;
+      }
+    }
+    .flatpickr-day.selected, .flatpickr-day.startRange, .flatpickr-day.endRange {
+      background-color: $primary;
+      border-color: $primary;
+      box-shadow: $shadow_deep;
+      font-weight: 600;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:206]: ??

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 13]
```diff
@@ -0,0 +1,1018 @@
...
+    }
+    // had to use flat hex colors without alpha channels
+    // the shadows were overlapping and opacities were adding together in spots
+    .flatpickr-day.inRange {
+      background-color: #E5EEFA;
+      border: 2px;
+      box-shadow: -5px 0 0 #E5EEFA, 5px 0 0 #E5EEFA;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:213]: tokens

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 14]
```diff
@@ -0,0 +1,1018 @@
...
+    }
+  }
+
+  &.error {
+    [class^=pb_body_kit] {
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:218]: can we use a class instead?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 15]
```diff
@@ -0,0 +1,1018 @@
...
+    .flatpickr-wrapper {
+      > input:first-child {
+        border-color: $error;
+      }
+    }
+  }
+}
+
+.pb_popover_body {
+  overflow: visible !important;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:230]: is the important needed?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 16]
```diff
@@ -0,0 +1,1018 @@
...
+  text-align: center;
+  visibility: hidden;
+  padding: 0;
+  -webkit-animation: none;
+          animation: none;
+  direction: ltr;
+  border: 0;
+  font-size: 14px;
+  line-height: 24px;
+  border-radius: 5px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:248]: tokens?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 17]
```diff
@@ -0,0 +1,1018 @@
...
+  position: absolute;
+  width: 307.875px;
+  -webkit-box-sizing: border-box;
+          box-sizing: border-box;
+  -ms-touch-action: manipulation;
+      touch-action: manipulation;
+  background: #fff;
+  -webkit-box-shadow: 1px 0 0 #e6e6e6, -1px 0 0 #e6e6e6, 0 1px 0 #e6e6e6, 0 -1px 0 #e6e6e6, 0 3px 13px rgba(0,0,0,0.08);
+          box-shadow: 1px 0 0 #e6e6e6, -1px 0 0 #e6e6e6, 0 1px 0 #e6e6e6, 0 -1px 0 #e6e6e6, 0 3px 13px rgba(0,0,0,0.08);
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:257]: tokens

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 18]
```diff
@@ -0,0 +1,1018 @@
...
+  font-size: 14px;
+  line-height: 24px;
+  border-radius: 5px;
+  position: absolute;
+  width: 307.875px;
+  -webkit-box-sizing: border-box;
+          box-sizing: border-box;
+  -ms-touch-action: manipulation;
+      touch-action: manipulation;
+  background: #fff;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:255]: token

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 19]
```diff
@@ -0,0 +1,1018 @@
...
+  -webkit-box-shadow: 1px 0 0 #e6e6e6, -1px 0 0 #e6e6e6, 0 1px 0 #e6e6e6, 0 -1px 0 #e6e6e6, 0 3px 13px rgba(0,0,0,0.08);
+          box-shadow: 1px 0 0 #e6e6e6, -1px 0 0 #e6e6e6, 0 1px 0 #e6e6e6, 0 -1px 0 #e6e6e6, 0 3px 13px rgba(0,0,0,0.08);
+}
+.flatpickr-calendar.open,
+.flatpickr-calendar.inline {
+  opacity: 1;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:261]: token

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 20]
```diff
@@ -0,0 +1,1018 @@
...
+  max-height: 640px;
+  visibility: visible;
+}
+.flatpickr-calendar.open {
+  display: inline-block;
+  z-index: 99999;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:267]: token? Also what z-index do we need in nitro. Will this display as intended?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 21]
```diff
@@ -0,0 +1,1018 @@
...
+          animation: fpFadeInDown 300ms cubic-bezier(0.23, 1, 0.32, 1);
+}
+.flatpickr-calendar.inline {
+  display: block;
+  position: relative;
+  top: 2px;
+}
+.flatpickr-calendar.static {
+  position: absolute;
+  top: calc(100% + 2px);
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:280]: Could this cause a problem in certain layouts? Are you expecting 100% to be the width of the screen or the parent container?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 22]
```diff
@@ -0,0 +1,1018 @@
...
+}
+.flatpickr-calendar.static.open {
+  z-index: 999;
+  display: block;
+}
+.flatpickr-calendar.multiMonth .flatpickr-days .dayContainer:nth-child(n+1) .flatpickr-day.inRange:nth-child(7n+7) {
+  -webkit-box-shadow: none !important;
+          box-shadow: none !important;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:288]: !important?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 23]
```diff
@@ -0,0 +1,1018 @@
...
+  border-bottom: 0;
+  border-bottom-right-radius: 0;
+  border-bottom-left-radius: 0;
+}
+.flatpickr-calendar .hasWeeks .dayContainer {
+  border-left: 0;
+}
+.flatpickr-calendar.showTimeInput.hasTime .flatpickr-time {
+  height: 40px;
+  border-top: 1px solid #e6e6e6;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:305]: token

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 24]
```diff
@@ -0,0 +1,1018 @@
...
+  z-index: 999;
+  display: block;
+}
+.flatpickr-calendar.multiMonth .flatpickr-days .dayContainer:nth-child(n+1) .flatpickr-day.inRange:nth-child(7n+7) {
+  -webkit-box-shadow: none !important;
+          box-shadow: none !important;
+}
+.flatpickr-calendar.multiMonth .flatpickr-days .dayContainer:nth-child(n+2) .flatpickr-day.inRange:nth-child(7n+1) {
+  -webkit-box-shadow: -2px 0 0 #e6e6e6, 5px 0 0 #e6e6e6;
+          box-shadow: -2px 0 0 #e6e6e6, 5px 0 0 #e6e6e6;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:292]: token

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 25]
```diff
@@ -0,0 +1,1018 @@
...
+/*
+      */
+}
+/*
+      /*rtl:begin:ignore*/
+/*
+      /*rtl:end:ignore*/
+.flatpickr-months .flatpickr-prev-month:hover,
+.flatpickr-months .flatpickr-next-month:hover {
+  color: #959ea9;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:439]: token

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 26]
```diff
@@ -0,0 +1,1018 @@
...
+}
+.flatpickr-months .flatpickr-prev-month:hover svg,
+.flatpickr-months .flatpickr-next-month:hover svg {
+  fill: #f64747;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:443]: token

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss (hunk 4 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.scss:100]: Agreed top area is pretty strange.  Flatpickr's div/tag structure is messy.  The wrapper for the selects is already `display: flex` but the month prev and month next arrows are `position: absolute`

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss (hunk 9 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.scss:169]: Switch to `@include pb_title_4`

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss (hunk 11 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.scss:196]: Switched to `font-weight: $bold;`.

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss (hunk 14 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.scss:218]: Changed to unique class name `.error-body-kit`

**Comment by thestephenmarshall** [comment]
@bh247484 lmk if you need to pair on getting the commits cleaned up with this one!

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 27]
```diff
@@ -0,0 +1,76 @@
...
+        transition: background-color 5000s ease-in-out 0s;
+      }
+      // Calendar Shadow and Border
+      .flatpickr-calendar {
+        box-shadow: $shadow_deeper;
+        border: 1px solid $border_light;
+      }
+      // Weekday styling
+      div.flatpickr-weekdays {
+        margin-top: 8px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:47]: Can we use a variable?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss [hunk 28]
```diff
@@ -0,0 +1,76 @@
+// @import "~flatpickr/dist/flatpickr.css";
+@import "../pb_caption/caption_mixin";
+@import "../pb_textarea/textarea_mixin";
+@import "../tokens/colors";
+@import "./sass_partials/flatpickr_styles";
+
+[class^=pb_date_picker_kit] {
+  // input styles
+  [class^=pb_caption_kit] {
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.scss:9]: Can we add a custom class to the caption kit so we don't have to do crazy class lookups?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.jsx [hunk 29]
```diff
@@ -0,0 +1,121 @@
...
+        <Caption
+            text={label}
+        />
+      </If>
+      <div className="input_wrapper">
+        <input
+            autoComplete="off"
+            id={pickerId}
+            name={name}
+        />
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.jsx:99]: Many of our form kits allow the user to bring their own input as they might need additional things added to the input that we don't enable through props. Do you think we would need that for this kit?

Comment (jasoncypret) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.jsx:99]: Follow up...do you think we could use the text input kit here instead?

File: app/pb_kits/playbook/pb_date_picker/_date_picker.html.erb [hunk 30]
```diff
@@ -0,0 +1,39 @@
...
+    <% if !object.hide_label %>
+      <%= pb_rails("caption", props: { text: object.label }) %>
+    <% end %>
+      <div class="input_wrapper">
+        <input
+            autocomplete="off"
+            id="<%= object.picker_id %>"
+            name="<%= object.name %>"
+            <%= object.required ? required="required" : ''%>
+        />
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/_date_picker.html.erb:15]: Many of our form kits allow the user to bring their own input as they might need additional things added to the input that we don't enable through props. Do you think we would need that for this kit?

Also, do you think we can use the text input kit here instead?

Comment (jasoncypret) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.html.erb:15]: How is dark passed to the input?

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss [hunk 31]
```diff
@@ -0,0 +1,47 @@
+// Day and Range Styles
+.flatpickr-day {
+  user-select: none;
+  @media (hover: hover) {
+    &:hover {
+      transition-duration: $transition_short;
+      transition-timing-function: $easeIn;    
+      background-color: #F6FAFF;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss:8]: Can this be a variable?

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
+}
+.flatpickr-day.selected, .flatpickr-day.startRange, .flatpickr-day.endRange {
+  border-radius: 50px;
+  background-color: $primary;
+  border-color: $primary;
+  box-shadow: $shadow_deep;
+  font-weight: $bold; 
+}
+.flatpickr-day.startRange {
+  box-shadow: 17px 0 0 #F7FBFF
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss:30]: Can we use our shadow variables?

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss [hunk 33]
```diff
@@ -0,0 +1,110 @@
...
+.flatpickr-months {
+  border-bottom: solid 1px $border_light;
+}
+.flatpickr-month {
+  height: $header_height;
+}
+.flatpickr-current-month {
+  left: 10px;
+  display: flex;
+  padding-top: 0px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss:14]:
```suggestion
  padding-top: 0;
```

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss [hunk 34]
```diff
@@ -0,0 +1,110 @@
...
+  display: contents;
+  .numInput {
+    @include pb_title_4;
+    border: 0px;
+    appearance: none;
+    background-color: transparent;
+    border-left: solid 1px $border_light;
+    padding-left: 22.5px;
+    padding-right: 14px;
+    margin-left: 10px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss:42]: Can we use spacing variables?

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss [hunk 35]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  }
+}
+// Icon for select tags
+.month-dropdown-icon {
+  align-self: center;
+  margin-top: 5px;
+  width: 10px;
+  position: relative;
+  right: 10px;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss:60]: Spacing variables?

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss [hunk 36]
```diff
@@ -0,0 +1,37 @@
+// media query non mobile styles
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss:1]: Lots of hardcoded widths here. Any concerns with resizing?

**Comment by jasoncypret** [comment]
One last thing. The calendar numbers jump on hover. It looks like this is because the numbers have a transparent border that is removed on hover.

image

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss [hunk 37]
```diff
@@ -0,0 +1,37 @@
...
+@media (min-width: 500px) {
+  .flatpickr-calendar {
+    padding: 8px;
```
[the remark this answers is not in the document]
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss:4]: Switched to `$space_xs`.

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss [hunk 38]
```diff
@@ -0,0 +1,37 @@
...
+    width: 323.88px
+  }
+  .dayContainer {
+    width: 315.88px
+  }
+  .flatpickr-months {
+    margin-left: -8px;
+    margin-top: -8px;
```
[the remark this answers is not in the document]
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss:12]: Only way I could get things re-aligned after adding padding.  Tried many other solutions.  Tricky because I had to work with whatever divs flatpickr already had templated instead of being able to rearrange them myself.

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss (hunk 36 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_overrides.scss:1]: Widths were added to accommodate extra padding.  Wanted to remove padding on smaller screens to maintain mobile usability.  Works as expected on any size screen.

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss (hunk 31 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss:8]: Had originally used variable noted in the mocks, `$neutral` I think.  Last round of styling feedback ask to match colors on mock so I grabbed the hex values manually.  Mock color value was not `$neutral`.  Can change back to `$neutral` if need be.

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss (hunk 27 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.scss]: Switched to `$space_xs`.

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss (hunk 32 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss:30]: These have to be manual values.  They are the colors generating the range highlighting.  I used our shadow mixins/variables for the day hovers.

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss [hunk 39]
```diff
@@ -0,0 +1,47 @@
...
+}
+.flatpickr-day.endRange {
+  box-shadow: -16px 0 0 #F7FBFF
```
[the remark this answers is not in the document]
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss:33]: These have to be manual values.  They are the colors generating the range highlighting.  I used our shadow mixins/variables for the day hovers.

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss [hunk 40]
```diff
@@ -0,0 +1,47 @@
...
+}
+// had to use flat hex colors without alpha channels
+// the shadows were overlapping and opacities were adding together in spots
+.flatpickr-day.inRange {
+  background-color: #F7FBFF;
+  border: 2px;
+  box-shadow: -4px 0 0 #F7FBFF, 5px 0 0 #F7FBFF;
```
[the remark this answers is not in the document]
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_day_styles.scss:40]: These have to be manual values.  They are the colors generating the range highlighting.  I used our shadow mixins/variables for the day hovers.

File: app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss [hunk 41]
```diff
@@ -0,0 +1,110 @@
...
+.flatpickr-current-month {
+  left: 10px;
+  display: flex;
+  padding-top: 0px;
+}
+.flatpickr-monthDropdown-months {
+  appearance: none;
+  @include pb_title_4;
+  height: $header_height;
+  width: 93px;
```
[the remark this answers is not in the document]
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/sass_partials/_header_styles.scss:20]: No, its the width of the month select.

File: app/pb_kits/playbook/pb_date_picker/_date_picker.scss (hunk 28 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.scss:9]: Added `.caption_selector` class name to caption kit.

**Comment by bh247484** [comment]
> One last thing. The calendar numbers jump on hover. It looks like this is because the numbers have a transparent border that is removed on hover.
> 
> image

Nice catch.  Had added `border: none;` since the default styling had a different color.  Matched the border color with the background color instead.  Number jump is gone.

File: app/pb_kits/playbook/pb_date_picker/_date_picker.html.erb (hunk 30 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.html.erb:15]: Removed html input tag and used text input kit instead.  Dark is now passed as a prop to the text input kit.

File: app/pb_kits/playbook/pb_date_picker/_date_picker.jsx (hunk 29 above)
Comment (bh247484) [reply app/pb_kits/playbook/pb_date_picker/_date_picker.jsx:103]: Removed html input tag and used text input kit instead.  Dark is now passed as a prop to the text input kit.

**Review by jasoncypret — APPROVED** [review]

**Review by jasoncypret — APPROVED** [review]
